cmd/gcp-controller-manager/app/gke_approver.go

```diff
@@ -425,38 +411,30 @@ func validateTPMAttestation(opts GCPConfig, csr *capi.CertificateSigningRequest,
 		glog.Errorf("Failed extracting VM identity from EK certificate: %v", err)
 		return false
 	}
-	hostname := strings.TrimPrefix("system:node:", x509cr.Subject.CommonName)
+	hostname := strings.TrimPrefix(x509cr.Subject.CommonName, "system:node:")
```

Comment:
Do we have a unit test that will catch this?

Reply:
I'm working on separate PR with tests. This entire func is currently untested :/